provider/kubernetes/client.go

```diff
@@ -150,31 +124,28 @@ func (c *clientImpl) WatchAll(namespaces Namespaces, labelSelector string, stopC
 		c.isNamespaceAll = true
 	}
 
-	var informManager informerManager
-	for _, ns := range namespaces {
-		ns := ns
-		informManager.extend(c.WatchIngresses(ns, kubeLabelSelector, eventCh), true)
-		informManager.extend(c.WatchObjects(ns, kindServices, &corev1.Service{}, c.svcStores, eventCh), true)
-		informManager.extend(c.WatchObjects(ns, kindEndpoints, &corev1.Endpoints{}, c.epStores, eventCh), true)
-		// Do not wait for the Secrets store to get synced since we cannot rely on
-		// users having granted RBAC permissions for this object.
-		// https://github.com/containous/traefik/issues/1784 should improve the
-		// situation here in the future.
-		informManager.extend(c.WatchObjects(ns, kindSecrets, &corev1.Secret{}, c.secStores, eventCh), false)
```

Comment:
We pass `false` as the last parameter so that we don't wait for the Secrets store to become synchronized for the described reason.

Am I missing something, or did we not carry over that behavior?

Reply:
Was this addressed?

Reply:
Yes and we skip the synchronization of secret by delaying the invocation of `AddEventHandler`.